bowtie/schemas/implementation-matrix-versions.json

```diff
@@ -0,0 +1,17 @@
+{
+  "title": "Implementation Matrix Versions",
+  "description": "Versions of an implementation that we are allowed to build images for.",
+
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+
+  "$id": "tag:bowtie.report,2024:implementation-matrix-versions",
+
+  "type": "array",
+
+  "items": {
+    "type": "string",
+    "pattern": "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)$"
```

Comment:
Use the field we already have for implementation versions, it's here:

https://github.com/bowtie-json-schema/bowtie/blob/main/bowtie/schemas/models/implementation.json#L27

We don't currently restrict things at all to a regex (it may not even be possible to, I'm not sure what regex covers every version format across all languages).

But yeah use a `$ref` to that (add an `$id` there probably to do so) and then we can independently decide whether to add a `pattern` or not.